### PR TITLE
refactor!: fully switch to Diataxis system

### PR DESCRIPTION
**Pull request by ahal**
BREAKING CHANGE: This will break pretty much every link to our docs.

Awhile back we agreed to try and implement the Diataxis documentation system:
https://diataxis.fr/

However, we only moved a handful of docs over and have been in a mixed state of having docs following two separate structures (if you can call the old way a structure) ever since.

The goal of this PR is to re-organize all the docs in the old structured over to the new Diataxis-based structure. There are plenty of docs that don't fit neatly in one place or the other, so this PR often takes a best guess at where it belongs.

This PR also doesn't split up any individual files, though it does sometimes move docs from directories to different places.

[Rendered](https://moz-releng-docs--240.org.readthedocs.build/en/240/index.html)

### CODE REVIEW

File: reference/index.rst [hunk 1]
```diff
@@ -2,20 +2,10 @@ Reference
...
-   Releng Mana <https://mana.mozilla.org/wiki/display/RelEng/Release+Engineering+Home>
-   Scriptworker <https://scriptworker.readthedocs.io/en/latest/index.html>
-   Scriptworker Scripts <https://scriptworker-scripts.readthedocs.io/en/latest/index.html>
-   Scriptworker-scripts Wiki (mac signing procedures) <https://github.com/mozilla-releng/scriptworker-scripts/wiki>
-   Taskgraph <https://taskcluster-taskgraph.readthedocs.io/>
-   Treeherder <https://treeherder.readthedocs.io/>
+   access
+   machine-users
+   logs
+   /releng_changelog
```
Comment (bhearsum) [inline reference/index.rst:10]: Is it time to do away with the changelog? If not, we should start using this again :)

File: index.rst [hunk 2]
```diff
@@ -1,57 +1,20 @@
-.. RelEng Docs documentation master file, created by
```
Comment (bhearsum) [inline index.rst:1]: This page ends up very lengthy, but I really, really, like how scannable, searchable it is. It's now much easier to get a broad overview of what's in here.

File: tutorials/mobile_automation_setup.rst [hunk 3]
```diff
@@ -1,5 +1,5 @@
-How to automate nightly Google Play deployments
-===============================================
+Automate Google Play Deployments
```
Comment (bhearsum) [inline tutorials/mobile_automation_setup.rst:1]: Is this a Tutorial or HOWTO? It reads more task-oriented than learning-oriented to me.

**Review by bhearsum — COMMENTED** [review]
I mostly looked over this for how things were oragnized. It looks excellent overall! Just a few minor things noted below, mostly questions.

File: reference/index.rst (hunk 1 above)
Comment (ahal) [reply reference/index.rst:10]: A very good question. Personally I've never found value in it even when we actually used it somewhat. This is probably worth its own bullet point at the team meeting.

**Comment by gbrownmozilla** [comment]
lgtm - thanks! 

I think we should make a pass at updating links in all the places we can think of.

We should discuss the changelog; I'm leaning toward retiring it - I haven't found it very useful, and we haven't been keeping it up to date.

**Comment by gabrielBusta** [comment]
Some redirects that might be nice to have:

- https://docs.mozilla-releng.net/en/latest/procedures/release-duty/
- https://moz-releng-docs.readthedocs.io/en/latest/taskcluster/ci_admin.html
- https://moz-releng-docs.readthedocs.io/en/latest/how-to/release/mozillavpn.html

**Review by bhearsum — APPROVED** [review]

**Comment by ahal** [comment]
(fixed merge conflicts)

**Pull request merged**